test/e2e/network_segmentation.go

```diff
@@ -676,6 +676,106 @@ var _ = Describe("Network Segmentation", func() {
+				withNodeSelector(map[string]string{nodeHostnameKey: workerTwoNodeName}),
+			)
+
+			By("creating second namespace")
+			_, err := cs.CoreV1().Namespaces().Create(context.Background(), &v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: netConfig2.namespace,
+				},
+			}, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			// required so the namespaces get cleaned up
+			defer func() {
+				Expect(cs.CoreV1().Namespaces().Delete(context.Background(), netConfig2.namespace, metav1.DeleteOptions{})).To(Succeed())
+			}()
+
+			By(fmt.Sprintf("creating the network in namespace %s", netConfig1.namespace))
+			udnManifest := generateUserDefinedNetworkManifest(&netConfig1)
+			cleanup, err := createManifest(netConfig1.namespace, udnManifest)
+			Expect(err).NotTo(HaveOccurred())
+			DeferCleanup(cleanup)
+			Expect(waitForUserDefinedNetworkReady(netConfig1.namespace, netConfig1.name, 5*time.Second)).To(Succeed())
+
+			By(fmt.Sprintf("creating client/server pods in namespace %s", netConfig1.namespace))
+			serverPodConfig.namespace = netConfig1.namespace
+			clientPodConfig.namespace = netConfig1.namespace
+			runUDNPod(cs, netConfig1.namespace, serverPodConfig, nil)
+			runUDNPod(cs, netConfig1.namespace, clientPodConfig, nil)
+
+			By(fmt.Sprintf("creating the network in namespace %s", netConfig2.namespace))
+			udnManifest = generateUserDefinedNetworkManifest(&netConfig2)
+			cleanup2, err := createManifest(netConfig2.namespace, udnManifest)
+			Expect(err).NotTo(HaveOccurred())
+			DeferCleanup(cleanup2)
+			Expect(waitForUserDefinedNetworkReady(netConfig2.namespace, netConfig2.name, 5*time.Second)).To(Succeed())
+
+			By(fmt.Sprintf("creating client/server pods in namespace %s", netConfig2.namespace))
+			serverPodConfig.namespace = netConfig2.namespace
+			clientPodConfig.namespace = netConfig2.namespace
+			runUDNPod(cs, netConfig2.namespace, serverPodConfig, nil)
+			runUDNPod(cs, netConfig2.namespace, clientPodConfig, nil)
+
+			var serverIP string
+			for _, config := range []networkAttachmentConfigParams{netConfig1, netConfig2} {
+				serverPodConfig.namespace = config.namespace
+				clientPodConfig.namespace = config.namespace
+				By(fmt.Sprintf("asserting network works in namespace %s", config.namespace))
+				for i, cidr := range strings.Split(config.cidr, ",") {
+					if cidr != "" {
+						By("asserting the server pod has an IP from the configured range")
+						serverIP, err = podIPsForUserDefinedPrimaryNetwork(
+							cs,
+							config.namespace,
+							serverPodConfig.name,
+							namespacedName(config.namespace, config.name),
+							i,
+						)
+						Expect(err).NotTo(HaveOccurred())
+						const netPrefixLengthPerNode = 24
+						By(fmt.Sprintf("asserting the server pod IP %v is from the configured range %v/%v", serverIP, cidr, netPrefixLengthPerNode))
+						subnet, err := getNetCIDRSubnet(cidr)
+						Expect(err).NotTo(HaveOccurred())
+						Expect(inRange(subnet, serverIP)).To(Succeed())
```

Comment:
Please note that checking pod got IP from spesifed CIDR it is already covered in other tests, see "creates a networkStatus Annotation with UDN interface" & "can perform east/west traffic between nodes" tests 

I seems that checking connectivity between the pod over the IP address from OVN annotation should do, WDYT?

Reply:
I just copied it from https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/test/e2e/network_segmentation.go#L158-L178 :D
I guess we can skip this check